ledger/accountdb_test.go

```diff
@@ -2118,6 +2165,143 @@ func TestBaseAccountDataIsEmpty(t *testing.T) {
 
 }
 
+func TestBaseOnlineAccountDataIsEmpty(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	positiveTesting := func(t *testing.T) {
+		var ba baseOnlineAccountData
+		require.True(t, ba.IsEmpty())
+		require.True(t, ba.IsVotingEmpty())
+		ba.MicroAlgos.Raw = 100
+		require.True(t, ba.IsVotingEmpty())
+		ba.RewardsBase = 200
+		require.True(t, ba.IsVotingEmpty())
+	}
+	var empty baseOnlineAccountData
+	negativeTesting := func(t *testing.T) {
+		for i := 0; i < 10; i++ {
+			randObj, _ := protocol.RandomizeObjectField(&baseOnlineAccountData{})
+			ba := randObj.(*baseOnlineAccountData)
+			if *ba == empty {
+				continue
+			}
+			require.False(t, ba.IsEmpty(), "base account : %v", ba)
+			break
+		}
+		{
+			var ba baseOnlineAccountData
+			ba.MicroAlgos.Raw = 100
+			require.False(t, ba.IsEmpty())
+		}
+		{
+			var ba baseOnlineAccountData
+			ba.RewardsBase = 200
+			require.False(t, ba.IsEmpty())
+		}
+	}
+	structureTesting := func(t *testing.T) {
+		encoding, err := json.Marshal(&empty)
+		expectedEncoding := `{"VoteID":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"SelectionID":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"VoteFirstValid":0,"VoteLastValid":0,"VoteKeyDilution":0,"StateProofID":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"MicroAlgos":{"Raw":0},"RewardsBase":0}`
```

Comment:
Would it be cleaner to use make([]byte, 32) to specify these empty arrays?

Reply:
these are string constants, well, maybe at least break to subcomps

Reply:
https://github.com/algorand/go-algorand/pull/4211